test/integration/server0_p_10_and_later_test.exs

```diff
@@ -24,12 +24,14 @@ defmodule KafkaEx.Server0P10P1AndLater.Test do
       Enum.member?(existing_topics(), name)
     end)
 
-    assert @num_partitions ==
-             KafkaEx.Protocol.Metadata.Response.partitions_for_topic(
-               KafkaEx.metadata(),
-               name
-             )
-             |> Enum.count()
+    wait_for(fn ->
```

Comment:
This will explicitly raise an error if it the comparison does not return `true` which fails the test

```
  1) test can create a topic (KafkaEx.Server0P10P1AndLater.Test)
     test/integration/server0_p_10_and_later_test.exs:9
     ** (RuntimeError) too many tries waiting for condition
     code: wait_for(fn ->
     stacktrace:
       test/test_helper.exs:165: TestHelper.wait_for_value/5
       test/integration/server0_p_10_and_later_test.exs:27: (test)
```